.buildkite/commands/test-android.sh

```diff
@@ -16,9 +16,7 @@ while [ "$INPUT" != "" ]; do
     INPUT="${1-}"
 done
 
-echo '--- :node: Set up Node depenendencies'
-npm ci --prefer-offline --no-audit --ignore-scripts
-npm ci --prefix gutenberg --prefer-offline --no-audit
+.buildkite/commands/install-node-dependencies.sh
```

Comment:
Looking at this and the CI logs. I think this might be a regression 😞 

Fingers crossed it should be a matter of replicating the options like:

```sh
# First, set up the gutenberg-mobile dependencies without the i18n cache (--ignore-scripts) which takes time and we don't need at this point as we are running the tests on top of something already built
.buildkite/commands/install-node-dependencies.sh --ignore-scripts
# Then, set up the gutenberg submodule dependencies, bypassed by the step above, which we need because some E2E logic lives there
.buildkite/commands/install-node-dependencies.sh --prefix gutenberg

Reply:
Still not perfect after adopting the split call approach, because the caches are downloaded twice.

<img width="1124" alt="image" src="https://github.com/wordpress-mobile/gutenberg-mobile/assets/1218433/852f6ff1-71bc-419f-bfce-6fb4f82bae74">

All in all, I think ~30s is acceptable at this stage. But, an idea to push this further would be to further divide `install-node-dependencies.sh` in

- `restore-node-cache.sh`
- command to install dependencies
- `save-node-cache.sh`

`install-node-dependencies.sh` could call all three steps. Scripts like this, which need to run multiple `npm` calls, drill down and only call `restore-node-cache.sh` (don't want to risk saving a partial cache)

Reply:
Done via [`50a3471` (#6707)](https://github.com/wordpress-mobile/gutenberg-mobile/pull/6707/commits/50a347147f7465da3baa99c93f003e8831bc1578)